config/7.0/obdemo-bank/ig/scripts/groovy/RCSGetConsent.groovy

```diff
@@ -1,110 +1,99 @@
+                    "identification" : null,
+                    "accountId" : "22289",
+                    "schemeName" : null,
+                    "secondaryIdentification" : null,
+                    "accountType" : "Personal",
+                    "accountSubType" : "CurrentAccount",
+                    "nickname" : "Bills",
+                    "name" : null
+            ],
+            [
+                    "currency" : "GBP",
+                    "account" : [ [
+                                          "currency" : null,
+                                          "account" : null,
+                                          "identification" : "80200110203348",
+                                          "accountId" : null,
+                                          "schemeName" : "UK.OBIE.SortCodeAccountNumber",
+                                          "secondaryIdentification" : null,
+                                          "accountType" : null,
+                                          "accountSubType" : null,
+                                          "nickname" : null,
+                                          "name" : "Mr Kevin"
+                                  ] ],
+                    "identification" : null,
+                    "accountId" : "31820",
+                    "schemeName" : null,
+                    "secondaryIdentification" : null,
+                    "accountType" : "Personal",
+                    "accountSubType" : "CurrentAccount",
+                    "nickname" : "Household",
+                    "name" : null
+            ]
     ]
 }
 
-// Pull consent details from intent id embedded in consent request from AM
-
-def consentRequest = contexts.jwtValidation.value
-def intentId = contexts.jwtValidation.claims.claims.id_token.openbanking_intent_id.value.toString().replaceAll('"','')
-def username = contexts.jwtValidation.claims.username.toString()
-def clientName = contexts.jwtValidation.claims.client_name.toString()
-
-def scopes = contexts.jwtValidation.claims.scopes
-def scopeList = scopes.asMap().entrySet().toArray()
-
-
-// Figure out whether this is an account info or payment request from the scope
-
-def flow = ""
+/*
+ * legacyAccountList
+ *
+ * Convert RS account data to old format for legacy UI
+ */
 
-if (scopes.contains("accounts"))  {
-    logger.debug("Detected AISP flow")
-  flow = "aisp"
-}
-else if (scopes.contains("payments")) {
-    logger.debug("Detected PISP flow")
-    flow = "pisp"
+def legacyAccountList(accountData) {
+
+    def accountList = []
+
+    accountData.accountDatas.forEach( data -> {
+            accountList.push (      [
+                    "currency" : data.account.Currency,
+                    "account" : [ [
+                                          "currency" : null,
+                                          "account" : null,
+                                          "identification" : data.account.Account[0].Identification,
```

Comment:
Will `data.account` or `data.account.Account` ever be null?

Reply:
They shouldn't be - but in any case, this is disposable code, which will be replaced in its entirety by the actual RCS service.